src/arwave.py

```diff
@@ -1,122 +1,214 @@
-
-    #else:
-        # tell code to continue when archived and object wavelengths are the
-        # same length? do I need to?
-
-
-    #cross correlate the two spectra, which are now on the same wavelength
-    #range and have the same array sizes
-    corr=np.correlate(arx_sky.flux,obj_sky.flux,"same")
-    plt.plot(corr)
-
-    #create an array around the max of the correlation function
-    max_corr=np.argmax(corr)
-    subpix_grid=np.linspace(max_corr-4,max_corr+4,9.)
-
-    #fit a gaussian to where the correlation function peaks
-    fit,other=arutils.gauss_lsqfit(subpix_grid,corr[subpix_grid.astype(np.int)],max_corr)
-
-    #calculate the shift and apply it to the wavelength
-    shift=fit[1]-0.
-    #model=fit[0]*(np.exp((-(subpix_grid-fit[1])**2)/(2*fit[2]**2)))
-    #plt.plot(subpix_grid,model)
-
-    finer_subpix_grid=np.linspace(max_corr-4,max_corr+4,90.)
-    model=fit[0]*(np.exp((-(finer_subpix_grid-fit[1])**2)/(2*fit[2]**2)))
-    plt.plot(finer_subpix_grid,model)
-    #plt.show()
-
-    return
-
-#flexure(object_wave2,object_flux)
+        skyspec_fil = slf._argflag['reduce']['flexure']['archive_spec']
+    msgs.info("Using {:s} file for Sky spectrum".format(skyspec_fil))
+    hdu = fits.open(root+'/data/sky_spec/'+skyspec_fil)
+    archive_wave = hdu[0].data
+    archive_flux = hdu[1].data
+    arx_sky = xspectrum1d.XSpectrum1D.from_tuple((archive_wave, archive_flux))
+
+    # Loop on objects
+    flex_dict = dict(polyfit=[], roots=[], shift=[], subpix=[], corr=[],
+                     corr_cen=[], spec_file=skyspec_fil, smooth=[],
+                     arx_spec=[], sky_spec=[])
+
+    for specobj in slf._specobjs[det-1]:  # for convenience
+
+        # Using boxcar
+        if slf._argflag['reduce']['flexure']['spec'] == 'boxcar':
+            sky_wave = specobj.boxcar['wave'].to('AA').value
+            sky_flux = specobj.boxcar['sky']
+        else:
+            msgs.error("Not ready for this flexure method: {}".format(
+                    slf._argflag['reduce']['flexure']))
+
+        # Generate 1D spectrum for object
+        obj_sky = xspectrum1d.XSpectrum1D.from_tuple((sky_wave, sky_flux))
+
+        #Determine the brightest emission lines
+        msgs.warn("If we use Paranal, cut down on wavelength early on")
+        arx_amp, arx_cent, arx_wid, arx_w, arx_satsnd, arx_yprep = aarc.detect_lines(slf, det, msarc=None, censpec=arx_sky.flux.value, MK_SATMASK=False)
+        obj_amp, obj_cent, obj_wid, obj_w, obj_satsnd, obj_yprep = aarc.detect_lines(slf, det, msarc=None, censpec=obj_sky.flux.value, MK_SATMASK=False)
+
+        #Keep only 5 brightest amplitude lines (xxx_keep is array of indices within arx_w of the 5 brightest)
+        arx_keep = np.argsort(arx_amp[arx_w])[-5:]
+        obj_keep = np.argsort(obj_amp[obj_w])[-5:]
+
+        #Calculate wavelength (Angstrom per pixel)
+        arx_disp = (np.amax(arx_sky.wavelength.value)-np.amin(arx_sky.wavelength.value))/arx_sky.wavelength.size
```

Comment:
Angstroms/pixel at every point in the wavelength array:
arx_disp = np.append(arx_sky.wavelength.value[1]-arx_sky.wavelength.value[0], arx_sky.wavelength.value[1:]-arx_sky.wavelength.value[:-1])